src/nft/components/sell/select/SelectPage.tsx

```diff
@@ -1,3 +1,380 @@
+      getNextPageParam: (lastPage, pages) => {
+        return lastPage?.flat().length === 25 ? pages.length : null
+      },
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+    }
+  )
+
+  const ownerAssets = useMemo(() => (isSuccess ? ownerAssetsData?.pages.flat() : null), [isSuccess, ownerAssetsData])
+
+  const walletAssets = useWalletCollections((state) => state.walletAssets)
+  const setWalletAssets = useWalletCollections((state) => state.setWalletAssets)
+  const displayAssets = useWalletCollections((state) => state.displayAssets)
+  const setDisplayAssets = useWalletCollections((state) => state.setDisplayAssets)
+  const setWalletCollections = useWalletCollections((state) => state.setWalletCollections)
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const reset = useSellAsset((state) => state.reset)
+  const setSellPageState = useSellPageState((state) => state.setSellPageState)
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    setWalletAssets(ownerAssets?.flat() ?? [])
+  }, [ownerAssets, setWalletAssets])
+
+  useEffect(() => {
+    ownerCollections && setWalletCollections(ownerCollections)
+  }, [ownerCollections, setWalletCollections])
+
+  const listFilter = useWalletCollections((state) => state.listFilter)
+
+  useEffect(() => {
+    if (searchText) {
+      const filtered = walletAssets.filter((asset) => asset.name?.toLowerCase().includes(searchText.toLowerCase()))
+      setDisplayAssets(filtered, listFilter)
+    } else {
+      setDisplayAssets(walletAssets, listFilter)
+    }
+  }, [searchText, walletAssets, listFilter, setDisplayAssets])
+
+  useEffect(() => {
+    setDisplayAssets(walletAssets, listFilter)
+  }, [walletAssets, listFilter, setDisplayAssets])
+
+  useEffect(() => {
+    if (ownerCollections?.length && collectionStats?.length) {
+      const ownerCollectionsCopy = [...ownerCollections]
+      for (const collection of ownerCollectionsCopy) {
+        const floorPrice = collectionStats.find((stat) => stat.address === collection.address)?.floorPrice
+        collection.floorPrice = floorPrice ? Math.round(floorPrice * 1000 + Number.EPSILON) / 1000 : 0 //round to at most 3 digits
+      }
+      setWalletCollections(ownerCollectionsCopy)
+    }
+  }, [collectionStats, ownerCollections, setWalletCollections])
+
+  return (
+    // Column style is temporary while we move over the filters bar that adjust width
+    <Column style={{ width: 'calc(100vw - 32px)' }}>
+      <Row
+        alignItems="flex-start"
+        position="relative"
+        paddingLeft={{ sm: '0', md: '52' }}
+        paddingRight={{ sm: '0', md: '72' }}
+        paddingTop={{ sm: '16', md: '40' }}
+      >
+        <AnimatedBox paddingX="16" flexShrink="0" width="full">
+          <Row gap="8" flexWrap="nowrap">
+            <CollectionSearch searchText={searchText} setSearchText={setSearchText} />
+            <SelectAllButton />
+          </Row>
+          <InfiniteScroll
```

Comment:
nit: could be clearer to follow by make a separate component called `AssetList`. same for `SellableAssetList` below.

imo best to keep Page components very high level around page structure and keep components isolated

Reply:
As a general design principle we kept related components colocated on a page provided the file didn't get overly bloated. If after our series of future architecture decisions we decided against this approach for all files, we can revisit afterwards.